sale_order_secondary_unit/report/sale_report_templates.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+<template id="report_saleorder_document" inherit_id="sale.report_saleorder_document">
+    <xpath expr="//t[@t-as='page']//th[2]" position="before">
+        <th class="text-right" groups="product_secondary_unit.group_secondary_unit">Second. Qty</th>
```

Comment:
I'm seeing this can have access problems. You have to put the group at template level (which is at the end a view), and not at field level).

Reply:
https://github.com/odoo/odoo/blob/22c6084688b7c67d93a1ba5962c2506d7a5b9262/odoo/addons/base/ir/ir_ui_view.py#L357